// @omega/core/src/const/trans-type.const.ts
/**
 *
 TRANS_TYPE -

 "UNKNOWN", "SALE", "RETURN", "AUTH", "POSTAUTH", "FORCEAUTH", "ADJUST", "WITHDRAWAL", "ACTIVATE",
 "ISSUE", "RELOAD", "CASHOUT", "DEACTIVATE", "REPLACE", "MERGE", "REPORTLOST", "VOID", "VOID SALE", "VOID RETURN",
 "VOID AUTH", "VOID POSTAUTH", "VOID FORCEAUTH", "VOID WITHDRAWAL", "INQUIRY", "VERIFY", "REACTIVATE", "FORCED ISSUE",
 "FORCED ADD", "UNLOAD", "RENEW", "TOKENIZE", "GETCONVERTDETAIL", "CONVERT", "INCREMENTALAUTH", "BALANCEWITHLOCK",
 "REDEMPTIONWITHUNLOCK", "REWARDS", "REENTER"
 */
export const TRANS_TYPE = {
  UNKNOWN: 'UNKNOWN',
  SALE: 'SALE',
  RETURN: 'RETURN',
  AUTH: 'AUTH',
  POST_AUTH: 'POST_AUTH',
  FORCE_AUTH: 'FORCE_AUTH',
  ADJUST: 'ADJUST',
  WITHDRAWAL: 'WITHDRAWAL',
  ACTIVATE: 'ACTIVATE',
  ISSUE: 'ISSUE',
  RELOAD: 'RELOAD',
  CASH_OUT: 'CASH_OUT',
  DEACTIVATE: 'DEACTIVATE',
  REPLACE: 'REPLACE',
  MERGE: 'MERGE',
  REPORT_LOST: 'REPORT_LOST',
  VOID: 'VOID',
  VOID_SALE: 'VOID_SALE',
  VOID_RETURN: 'VOID_RETURN',
  VOID_AUTH: 'VOID_AUTH',
  VOID_POST_AUTH: 'VOID_POST_AUTH',
  VOID_FORCE_AUTH: 'VOID_FORCE_AUTH',
  VOID_WITHDRAWAL: 'VOID_WITHDRAWAL',
  INQUIRY: 'INQUIRY',
  VERIFY: 'VERIFY',
  REACTIVATE: 'REACTIVATE',
  FORCED_ISSUE: 'FORCED_ISSUE',
  FORCED_ADD: 'FORCED_ADD',
  UNLOAD: 'UNLOAD',
  RENEW: 'RENEW',
  TOKENIZE: 'TOKENIZE',
  GET_CONVERT_DETAIL: 'GET_CONVERT_DETAIL',
  CONVERT: 'CONVERT',
  INCREMENTAL_AUTH: 'INCREMENTAL_AUTH',
  BALANCE_WITH_LOCK: 'BALANCE_WITH_LOCK',
  REDEMPTION_WITH_UNLOCK: 'REDEMPTION_WITH_UNLOCK',
  REWARDS: 'REWARDS',
  REENTER: 'REENTER',
  
};
