// @omega/core/dist/enums/country-code.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryCode = void 0;
var CountryCode;
(function (CountryCode) {
    CountryCode[CountryCode["AF"] = 0] = "AF";
    CountryCode[CountryCode["AL"] = 1] = "AL";
    CountryCode[CountryCode["DZ"] = 2] = "DZ";
    CountryCode[CountryCode["AD"] = 3] = "AD";
    CountryCode[CountryCode["AO"] = 4] = "AO";
    CountryCode[CountryCode["AQ"] = 5] = "AQ";
    CountryCode[CountryCode["AR"] = 6] = "AR";
    CountryCode[CountryCode["AM"] = 7] = "AM";
    CountryCode[CountryCode["AW"] = 8] = "AW";
    CountryCode[CountryCode["AU"] = 9] = "AU";
    CountryCode[CountryCode["AT"] = 10] = "AT";
    CountryCode[CountryCode["AZ"] = 11] = "AZ";
    CountryCode[CountryCode["BH"] = 12] = "BH";
    CountryCode[CountryCode["BD"] = 13] = "BD";
    CountryCode[CountryCode["BY"] = 14] = "BY";
    CountryCode[CountryCode["BE"] = 15] = "BE";
    CountryCode[CountryCode["BZ"] = 16] = "BZ";
    CountryCode[CountryCode["BJ"] = 17] = "BJ";
    CountryCode[CountryCode["BT"] = 18] = "BT";
    CountryCode[CountryCode["BO"] = 19] = "BO";
    CountryCode[CountryCode["BA"] = 20] = "BA";
    CountryCode[CountryCode["BW"] = 21] = "BW";
    CountryCode[CountryCode["BR"] = 22] = "BR";
    CountryCode[CountryCode["BN"] = 23] = "BN";
    CountryCode[CountryCode["BG"] = 24] = "BG";
    CountryCode[CountryCode["BF"] = 25] = "BF";
    CountryCode[CountryCode["MM"] = 26] = "MM";
    CountryCode[CountryCode["BI"] = 27] = "BI";
    CountryCode[CountryCode["KH"] = 28] = "KH";
    CountryCode[CountryCode["CM"] = 29] = "CM";
    CountryCode[CountryCode["CA"] = 30] = "CA";
    CountryCode[CountryCode["CV"] = 31] = "CV";
    CountryCode[CountryCode["CF"] = 32] = "CF";
    CountryCode[CountryCode["TD"] = 33] = "TD";
    CountryCode[CountryCode["CL"] = 34] = "CL";
    CountryCode[CountryCode["CN"] = 35] = "CN";
    CountryCode[CountryCode["CX"] = 36] = "CX";
    CountryCode[CountryCode["CC"] = 37] = "CC";
    CountryCode[CountryCode["CO"] = 38] = "CO";
    CountryCode[CountryCode["KM"] = 39] = "KM";
    CountryCode[CountryCode["CG"] = 40] = "CG";
    CountryCode[CountryCode["CD"] = 41] = "CD";
    CountryCode[CountryCode["CK"] = 42] = "CK";
    CountryCode[CountryCode["CR"] = 43] = "CR";
    CountryCode[CountryCode["HR"] = 44] = "HR";
    CountryCode[CountryCode["CU"] = 45] = "CU";
    CountryCode[CountryCode["CY"] = 46] = "CY";
    CountryCode[CountryCode["CZ"] = 47] = "CZ";
    CountryCode[CountryCode["DK"] = 48] = "DK";
    CountryCode[CountryCode["DJ"] = 49] = "DJ";
    CountryCode[CountryCode["TL"] = 50] = "TL";
    CountryCode[CountryCode["EC"] = 51] = "EC";
    CountryCode[CountryCode["EG"] = 52] = "EG";
    CountryCode[CountryCode["SV"] = 53] = "SV";
    CountryCode[CountryCode["GQ"] = 54] = "GQ";
    CountryCode[CountryCode["ER"] = 55] = "ER";
    CountryCode[CountryCode["EE"] = 56] = "EE";
    CountryCode[CountryCode["ET"] = 57] = "ET";
    CountryCode[CountryCode["FK"] = 58] = "FK";
    CountryCode[CountryCode["FO"] = 59] = "FO";
    CountryCode[CountryCode["FJ"] = 60] = "FJ";
    CountryCode[CountryCode["FI"] = 61] = "FI";
    CountryCode[CountryCode["FR"] = 62] = "FR";
    CountryCode[CountryCode["PF"] = 63] = "PF";
    CountryCode[CountryCode["GA"] = 64] = "GA";
    CountryCode[CountryCode["GM"] = 65] = "GM";
    CountryCode[CountryCode["GE"] = 66] = "GE";
    CountryCode[CountryCode["DE"] = 67] = "DE";
    CountryCode[CountryCode["GH"] = 68] = "GH";
    CountryCode[CountryCode["GI"] = 69] = "GI";
    CountryCode[CountryCode["GR"] = 70] = "GR";
    CountryCode[CountryCode["GL"] = 71] = "GL";
    CountryCode[CountryCode["GT"] = 72] = "GT";
    CountryCode[CountryCode["GN"] = 73] = "GN";
    CountryCode[CountryCode["GW"] = 74] = "GW";
    CountryCode[CountryCode["GY"] = 75] = "GY";
    CountryCode[CountryCode["HT"] = 76] = "HT";
    CountryCode[CountryCode["HN"] = 77] = "HN";
    CountryCode[CountryCode["HK"] = 78] = "HK";
    CountryCode[CountryCode["HU"] = 79] = "HU";
    CountryCode[CountryCode["IN"] = 80] = "IN";
    CountryCode[CountryCode["ID"] = 81] = "ID";
    CountryCode[CountryCode["IR"] = 82] = "IR";
    CountryCode[CountryCode["IQ"] = 83] = "IQ";
    CountryCode[CountryCode["IE"] = 84] = "IE";
    CountryCode[CountryCode["IM"] = 85] = "IM";
    CountryCode[CountryCode["IL"] = 86] = "IL";
    CountryCode[CountryCode["IT"] = 87] = "IT";
    CountryCode[CountryCode["CI"] = 88] = "CI";
    CountryCode[CountryCode["JP"] = 89] = "JP";
    CountryCode[CountryCode["JO"] = 90] = "JO";
    CountryCode[CountryCode["KZ"] = 91] = "KZ";
    CountryCode[CountryCode["KE"] = 92] = "KE";
    CountryCode[CountryCode["KI"] = 93] = "KI";
    CountryCode[CountryCode["KW"] = 94] = "KW";
    CountryCode[CountryCode["KG"] = 95] = "KG";
    CountryCode[CountryCode["LA"] = 96] = "LA";
    CountryCode[CountryCode["LV"] = 97] = "LV";
    CountryCode[CountryCode["LB"] = 98] = "LB";
    CountryCode[CountryCode["LS"] = 99] = "LS";
    CountryCode[CountryCode["LR"] = 100] = "LR";
    CountryCode[CountryCode["LY"] = 101] = "LY";
    CountryCode[CountryCode["LI"] = 102] = "LI";
    CountryCode[CountryCode["LT"] = 103] = "LT";
    CountryCode[CountryCode["LU"] = 104] = "LU";
    CountryCode[CountryCode["MO"] = 105] = "MO";
    CountryCode[CountryCode["MK"] = 106] = "MK";
    CountryCode[CountryCode["MG"] = 107] = "MG";
    CountryCode[CountryCode["MW"] = 108] = "MW";
    CountryCode[CountryCode["MY"] = 109] = "MY";
    CountryCode[CountryCode["MV"] = 110] = "MV";
    CountryCode[CountryCode["ML"] = 111] = "ML";
    CountryCode[CountryCode["MT"] = 112] = "MT";
    CountryCode[CountryCode["MH"] = 113] = "MH";
    CountryCode[CountryCode["MR"] = 114] = "MR";
    CountryCode[CountryCode["MU"] = 115] = "MU";
    CountryCode[CountryCode["YT"] = 116] = "YT";
    CountryCode[CountryCode["MX"] = 117] = "MX";
    CountryCode[CountryCode["FM"] = 118] = "FM";
    CountryCode[CountryCode["MD"] = 119] = "MD";
    CountryCode[CountryCode["MC"] = 120] = "MC";
    CountryCode[CountryCode["MN"] = 121] = "MN";
    CountryCode[CountryCode["ME"] = 122] = "ME";
    CountryCode[CountryCode["MA"] = 123] = "MA";
    CountryCode[CountryCode["MZ"] = 124] = "MZ";
    CountryCode[CountryCode["NA"] = 125] = "NA";
    CountryCode[CountryCode["NR"] = 126] = "NR";
    CountryCode[CountryCode["NP"] = 127] = "NP";
    CountryCode[CountryCode["NL"] = 128] = "NL";
    CountryCode[CountryCode["AN"] = 129] = "AN";
    CountryCode[CountryCode["NC"] = 130] = "NC";
    CountryCode[CountryCode["NZ"] = 131] = "NZ";
    CountryCode[CountryCode["NI"] = 132] = "NI";
    CountryCode[CountryCode["NE"] = 133] = "NE";
    CountryCode[CountryCode["NG"] = 134] = "NG";
    CountryCode[CountryCode["NU"] = 135] = "NU";
    CountryCode[CountryCode["KP"] = 136] = "KP";
    CountryCode[CountryCode["NO"] = 137] = "NO";
    CountryCode[CountryCode["OM"] = 138] = "OM";
    CountryCode[CountryCode["PK"] = 139] = "PK";
    CountryCode[CountryCode["PW"] = 140] = "PW";
    CountryCode[CountryCode["PA"] = 141] = "PA";
    CountryCode[CountryCode["PG"] = 142] = "PG";
    CountryCode[CountryCode["PY"] = 143] = "PY";
    CountryCode[CountryCode["PE"] = 144] = "PE";
    CountryCode[CountryCode["PH"] = 145] = "PH";
    CountryCode[CountryCode["PN"] = 146] = "PN";
    CountryCode[CountryCode["PL"] = 147] = "PL";
    CountryCode[CountryCode["PT"] = 148] = "PT";
    CountryCode[CountryCode["PR"] = 149] = "PR";
    CountryCode[CountryCode["QA"] = 150] = "QA";
    CountryCode[CountryCode["RO"] = 151] = "RO";
    CountryCode[CountryCode["RU"] = 152] = "RU";
    CountryCode[CountryCode["RW"] = 153] = "RW";
    CountryCode[CountryCode["BL"] = 154] = "BL";
    CountryCode[CountryCode["WS"] = 155] = "WS";
    CountryCode[CountryCode["SM"] = 156] = "SM";
    CountryCode[CountryCode["ST"] = 157] = "ST";
    CountryCode[CountryCode["SA"] = 158] = "SA";
    CountryCode[CountryCode["SN"] = 159] = "SN";
    CountryCode[CountryCode["RS"] = 160] = "RS";
    CountryCode[CountryCode["SC"] = 161] = "SC";
    CountryCode[CountryCode["SL"] = 162] = "SL";
    CountryCode[CountryCode["SG"] = 163] = "SG";
    CountryCode[CountryCode["SK"] = 164] = "SK";
    CountryCode[CountryCode["SI"] = 165] = "SI";
    CountryCode[CountryCode["SB"] = 166] = "SB";
    CountryCode[CountryCode["SO"] = 167] = "SO";
    CountryCode[CountryCode["ZA"] = 168] = "ZA";
    CountryCode[CountryCode["KR"] = 169] = "KR";
    CountryCode[CountryCode["ES"] = 170] = "ES";
    CountryCode[CountryCode["LK"] = 171] = "LK";
    CountryCode[CountryCode["SH"] = 172] = "SH";
    CountryCode[CountryCode["PM"] = 173] = "PM";
    CountryCode[CountryCode["SD"] = 174] = "SD";
    CountryCode[CountryCode["SR"] = 175] = "SR";
    CountryCode[CountryCode["SZ"] = 176] = "SZ";
    CountryCode[CountryCode["SE"] = 177] = "SE";
    CountryCode[CountryCode["CH"] = 178] = "CH";
    CountryCode[CountryCode["SY"] = 179] = "SY";
    CountryCode[CountryCode["TW"] = 180] = "TW";
    CountryCode[CountryCode["TJ"] = 181] = "TJ";
    CountryCode[CountryCode["TZ"] = 182] = "TZ";
    CountryCode[CountryCode["TH"] = 183] = "TH";
    CountryCode[CountryCode["TG"] = 184] = "TG";
    CountryCode[CountryCode["TK"] = 185] = "TK";
    CountryCode[CountryCode["TO"] = 186] = "TO";
    CountryCode[CountryCode["TN"] = 187] = "TN";
    CountryCode[CountryCode["TR"] = 188] = "TR";
    CountryCode[CountryCode["TM"] = 189] = "TM";
    CountryCode[CountryCode["TV"] = 190] = "TV";
    CountryCode[CountryCode["AE"] = 191] = "AE";
    CountryCode[CountryCode["UG"] = 192] = "UG";
    CountryCode[CountryCode["GB"] = 193] = "GB";
    CountryCode[CountryCode["UA"] = 194] = "UA";
    CountryCode[CountryCode["UY"] = 195] = "UY";
    CountryCode[CountryCode["US"] = 196] = "US";
    CountryCode[CountryCode["UZ"] = 197] = "UZ";
    CountryCode[CountryCode["VU"] = 198] = "VU";
    CountryCode[CountryCode["VA"] = 199] = "VA";
    CountryCode[CountryCode["VE"] = 200] = "VE";
    CountryCode[CountryCode["VN"] = 201] = "VN";
    CountryCode[CountryCode["WF"] = 202] = "WF";
    CountryCode[CountryCode["YE"] = 203] = "YE";
    CountryCode[CountryCode["ZM"] = 204] = "ZM";
    CountryCode[CountryCode["ZW"] = 205] = "ZW";
})(CountryCode = exports.CountryCode || (exports.CountryCode = {}));
